docs/versioning-policy.md

```diff
@@ -0,0 +1,70 @@
+# Upjet provider versioning and upgrade policy
+
+This policy recommends how Upjet providers should implement versioning numbers and handle upgrades to new versions.
+
+## Versioning 
+
+Upjet-based providers **MUST** follow the [semantic versioning](https://semver.org/) numbering scheme. 
+
+Specifically, a version contains 3 digits known as `major`, `minor` and `patch` numbers in the format `major.minor.patch` (e.g., `v1.2.3`).
+
+### Versioning specification
+
+1. The `major` version number **MUST** be incremented if the `major` version number of the Terraform provider it is generated from is incremented.
+2. The `minor` version number **MUST** be incremented when new functionality is released or when [unavoidable breaking changes](#unavoidable-breaking-changes) are introduced. 
+3. The `patch` version number **SHOULD ONLY** be incremented for a release containing **ONLY** bug fixes and no new features.
+4. A release that increments the `minor` or `patch` version **ONLY**, **MUST** be backward compatible.
+5. Incrementing the `minor` version **MUST** reset the `patch` version number to zero.
+6. Incrementing the `major` version **MUST** reset the `minor` and `patch` version number to zero.
+
+### Unavoidable breaking changes
+
+Due to a reliance on the upstream Terraform providers and prior instances where they introduce breaking changes in minor version releases, we cannot guarantee that a minor version bump will never introduce a breaking change. 
```

Comment:
I'm not familiar with the history here. Is the issue with terraform provider minor releases that have contained undocumented/surprising breaking changes? Or deliberate breaking changes, noted in the release notes, but released as a minor version bump. 

If the former, then this caveat is probably necessary. But if it's the latter, it seems we could just bump our major version. This would mean that the upjet provider's major version number would no longer match the terraform provider's major version number, but that seems like a reasonable price to pay for standard semver guarantees related to breaking changes.